Sources/LogInOut.php

```diff
@@ -97,21 +97,25 @@ function Login2()
 
 	if (isset($_GET['sa']) && $_GET['sa'] == 'salt' && !$user_info['is_guest'])
 	{
-		if (isset($_COOKIE[$cookiename]) && preg_match('~^a:[34]:\{i:0;i:\d{1,7};i:1;s:(0|128):"([a-fA-F0-9]{128})?";i:2;[id]:\d{1,14};(i:3;i:\d;)?\}$~', $_COOKIE[$cookiename]) === 1)
+		// First check for 2.1 json-format cookie in $_COOKIE
+		if (isset($_COOKIE[$cookiename]) && preg_match('~^\[\d{1,7},"([a-fA-F0-9]{128})?",\d{1,14}(,[0-3])?\]$~', $_COOKIE[$cookiename]) === 1)
```

Comment:
As fare my test goes the regex pattern didn't work for json...

Reply:
I've tested it repeatedly.  Works fine.

Reply:
http://www.phpliveregex.com/p/lEX here my test case

Reply:
Here is a screenshot of the point in question.  Note the format of the cookie.  Note also that, in the debugger, that line of code would not have been executed without a successful preg_match.

Note also that this is the logic for detecting cookie state changes - the heart of the original issue.  

![image](https://user-images.githubusercontent.com/23568484/31595528-4564755e-b1f1-11e7-9801-c05099f31f00.png)